crates/cheatcodes/src/test.rs

```diff
@@ -33,6 +36,23 @@ impl Cheatcode for breakpoint_1Call {
     }
 }
 
+impl Cheatcode for getFoundryVersionCall {
+    fn apply(&self, _state: &mut Cheatcodes) -> Result {
+        let Self {} = self;
+        let cargo_version = env!("CARGO_PKG_VERSION");
+        let build_timestamp = env::var("VERGEN_BUILD_TIMESTAMP")
+            .map(|ts| {
+                DateTime::parse_from_rfc3339(&ts)
+                    .expect("Invalid build timestamp format")
+                    .format("%Y%m%d%H%M")
+                    .to_string()
+            })
+            .unwrap_or_else(|_| "unknown".to_string());
+        let foundry_version = format!("{cargo_version}+{build_timestamp}");
```

Comment:
I think it would be best if this was moved outside of the cheatcode implementation, see
https://github.com/paradigmxyz/reth/blob/main/crates/node/core/src/version.rs so it can be re-used in other places and maintained more easily

Reply:
not that easy because then it has to be passed down to this crate. using the env vars here is fine imo, because we only expect this to be useful when used in forge binary